### PR TITLE
types: add failures and message to npm module result

### LINKED ISSUE #5335: Add optional properties returned from Cypress NPM module API on failure
If Cypress NPM module fails to find any specs it resolves with an object that is not typed right now

```js
// https://on.cypress.io/module-api
const cypress = require('cypress')
const results = await cypress.run()
// seems Cypress TS definition does not have "failures" and "message" properties
// @ts-ignore
if (results.failures) {
  // Cypress failed without even running the tests
  console.error('Problem running Cypress')
  // @ts-ignore
  console.error(results.message)
  // @ts-ignore
  process.exit(results.failures)
}

// results.totalFailed gives total number of failed tests
if (results.totalFailed) {
  process.exit(results.totalFailed)
}
```

### PR DESCRIPTION
**Pull request by bahmutov**
- Closes #5335 

### User facing changelog

Added TypeScript definition for two properties missing from the object returned from the Cypress module API run, see the issue for details

### PR Tasks

- [x] Have tests been added/updated?
- [x] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?

### CODE REVIEW

File: cli/types/cypress-npm-api.d.ts [hunk 1]
```diff
@@ -465,6 +466,31 @@ declare module 'cypress' {
     cypressVersion: string
     // TODO add resolved object to the configuration
     config: CypressConfiguration
+    /**
+     * If Cypress fails to run at all (for example, if there are no spec files to run),
+     * then it will set failures to 1 and will have actual error message in the
+     * property "message". Check this property before checking other properties.
+     *
+     * @type {number}
+     * @memberof CypressRunResult
```
Comment (flotwig) [inline cli/types/cypress-npm-api.d.ts]: is this necessary? i don't get why this is needed if it's already in the `interface` declaration

File: cli/types/cypress-npm-api.d.ts [hunk 2]
```diff
@@ -465,6 +466,31 @@ declare module 'cypress' {
...
+      }
+      ```
+     */
+    failures?: number
+    /**
+     * If returned result has "failures" set, then this property gives
+     * the error message.
+     *
+     * @type {string}
+     * @memberof CypressRunResult
```
Comment (flotwig) [inline cli/types/cypress-npm-api.d.ts]: ditto?

File: cli/types/cypress-npm-api.d.ts (hunk 1 above)
Comment (bahmutov) [reply cli/types/cypress-npm-api.d.ts]: that's how JSDoc "document this" extension does it, maybe it is unnecessary, let me remove it

**Review by flotwig — APPROVED** [review]
Looks good, is there a corresponding docs PR? The docs is missing these keys: https://docs.cypress.io/guides/guides/module-api.html#Example

**Comment by bahmutov** [comment]
Because the object is big and complex, in the docs we say the following
```
Find the TypeScript definition for the results object in the cypress/cli/types folder.
```
which links back to the types.

**Comment by flotwig** [comment]
> Because the object is big and complex, in the docs we say the following
> 
> ```
> Find the TypeScript definition for the results object in the cypress/cli/types folder.
> ```
> 
> which links back to the types.

If that's the case, the giant JSON blob should probably be removed, since it totally blocks the reader from seeing that bit of text at the end:

image

**Comment by bahmutov** [comment]
I am not sure it is that bad, plus gives a taste of what the object contains.

**Comment by flotwig** [comment]
It's out of date with the actual functionality so it should just be removed if it's not going to be updated. I did not even see the text you showed me until you pointed it out because I assumed this big block of JSON was up to date, else, why would it be in the docs?

**Pull request merged**